ui/pages/settings/networks-tab/networks-list/networks-list.js

```diff
@@ -9,22 +16,73 @@ const NetworksList = ({
   networkDefaultedToProvider,
   selectedRpcUrl,
 }) => {
+  const t = useI18nContext();
+  const [networks, setNetworks] = useState([]);
+  const [searchQuery, setSearchQuery] = useState('');
+  const networksToRenderProperly =
+    networks.length === 0 && searchQuery === '' ? networksToRender : networks;
+
   return (
     <div
       className={classnames('networks-tab__networks-list', {
         'networks-tab__networks-list--selection':
           networkIsSelected && !networkDefaultedToProvider,
       })}
     >
-      {networksToRender.map((network, index) => (
-        <NetworksListItem
-          key={`settings-network-list:${network.rpcUrl}`}
-          network={network}
-          networkIsSelected={networkIsSelected}
-          selectedRpcUrl={selectedRpcUrl}
-          networkIndex={index}
-        />
-      ))}
+      <CustomContentSearch
+        onSearch={({
+          searchQuery: newSearchQuery = '',
+          results: newResults = [],
+        }) => {
+          setNetworks(newResults);
+          setSearchQuery(newSearchQuery);
+        }}
+        error={
+          networksToRenderProperly.length === 0
+            ? t('settingsSearchMatchingNotFound')
+            : null
+        }
+        networksList={networksToRender}
+        searchQueryInput={searchQuery}
+      />
+      {networksToRenderProperly.map(
```

Comment:
Here we are using a map to filter out items, by producing an array of the form `[<ReactComponent />, false, <ReactComponent />, false, etc...]` React just ignores the `false` entries when rendering, but I think the purpose of the code would be clearer if a filter call, and perhaps a named variable, was used instead.

As a general rule, I think it is best if `map`s of arrays to components always return arrays of only components, and for the filtering to be done before the map call.

So I suggest adding something like the following above:
```
const searchedNetworksToRenderThatAreNotTestNetworks = networksToRenderProperly.filter(network => network.userIsCurrentlyOnATestNet === false);
const searchedNetworksToRenderThatAreTestNetworks = networksToRenderProperly.filter(network => network.userIsCurrentlyOnATestNet === true);
```
And then map those two variables/arrays as is done here, but now without the `network.userIsCurrentlyOnATestNet === false &&` and `network.userIsCurrentlyOnATestNet &&`

Reply:
@danjm Fixed in: 2179697b44f506d161e5258c52789f632ae8bda8